### ISSUE TITLE
User Commands

### ISSUE THREAD

**Issue by p1839251** [sahib/rmlint#325]
Hello,

I'm asking this because removing files can be really scary, and it's better to ask the way than go astray.
I don't quite understand the user commands part. The default remove command is `rm -rf`, isn't it?!
I wish to securely remove files by '_shred_' and use the `shred -vun 10` command this time.

Under formatters I found _cmd_ and _handler_. It's confusing because usercmd handler suggests `-c sh:cmd=something` and I would say this is the cmd option and not the handler.

So I'm not going to demonstrate how bad commands I've been managed to write so far, rather I'm asking you to tell me the right way to remove duplicates by `shred -vun 10`

Thank you in advance

**Comment by sahib** [comment]
Should be as simple as this:

```bash
# $1 is replaced with the path of the duplicate
rmlint -c 'sh:cmd=shred -vun 10 $1'
```

Can you please write a bit clearer what confused you in the documentation?

**Comment by p1839251** [comment]
Thank you for your answer

> Can you please write a bit clearer what confused you in the documentation?

Sure thing,
I think it's useful enough to deserve a separate example. It seems '-c sh:cmd=something' wasn't enough as I was trying (in this order): 
rmlint path -c sh:cmd="shred -vun 10" ; ./rmlint.sh -n
rmlint path -c sh:usercmd="shred -vun 10"; ./rmlint.sh -n
rmlint path -c sh:cmd="shred -vu -n 10 $1" ; ./rmlint.sh -n
rmlint path -c sh:handler=usercmd="shred -vun 10 $1"; ./rmlint.sh -n
rmlint path -c sh:handler=cmd="shred -vun 10 $1"; ./rmlint.sh -n

I think I gave up here. It's obvious that I don't understand the `usercmd` handler :/

./rmlint.sh -n kept telling me : shred: missing file operand

rmlint -c 'sh:cmd=shred -vun 10 $1' works if the files don't contain special characters, just like spaces:

`$ ls original/ duplicate/
duplicate/:
'file 1.mkv'

original/:
'file 0.mkv'

$ rmlint original/ duplicate/ -c 'sh:cmd=shred -vun 10 $1'  

Rmlint was executed in the following way:
   $ rmlint original/ duplicate/ -c sh:cmd=shred -vun 10 
Execute this script with -d to disable this informational message.
Type any string to continue; CTRL-C, Enter or CTRL-D to abort immediately
ok
[  0%] Keeping:  /tmp/original/file 0.mkv
[ 50%] shred: /tmp/duplicate/file: failed to open for writing: No such file or directory
shred: 1.mkv: failed to open for writing: No such file or directory
[100%] Done!
`
This should be fixed somehow :|

I also noticed if the file is completely regular (no special accented letters or spaces), the dry run `./rmlint -n` is rather wet: 

```
$ ./rmlint.sh -n
# ////////////////////////////////////////////////////////////
# ///  This is only a dry run; nothing will be modified! ///
# ////////////////////////////////////////////////////////////
[  0%] Keeping:  /tmp/original/file.mkv
[ 50%] shred: /tmp/duplicate/file.mkv: pass 1/10 (random)...
shred: /tmp/duplicate/file.mkv: pass 2/10 (555555)...
shred: /tmp/duplicate/file.mkv: pass 3/10 (b6db6d)...
shred: /tmp/duplicate/file.mkv: pass 4/10 (aaaaaa)...
shred: /tmp/duplicate/file.mkv: pass 5/10 (6db6db)...
shred: /tmp/duplicate/file.mkv: pass 6/10 (random)...
shred: /tmp/duplicate/file.mkv: pass 7/10 (ffffff)...
shred: /tmp/duplicate/file.mkv: pass 8/10 (db6db6)...
shred: /tmp/duplicate/file.mkv: pass 9/10 (000000)...
shred: /tmp/duplicate/file.mkv: pass 10/10 (random)...
shred: /tmp/duplicate/file.mkv: removing
shred: /tmp/duplicate/file.mkv: renamed to /tmp/duplicate/00000000
shred: /tmp/duplicate/00000000: renamed to /tmp/duplicate/0000000
shred: /tmp/duplicate/0000000: renamed to /tmp/duplicate/000000
shred: /tmp/duplicate/000000: renamed to /tmp/duplicate/00000
shred: /tmp/duplicate/00000: renamed to /tmp/duplicate/0000
shred: /tmp/duplicate/0000: renamed to /tmp/duplicate/000
shred: /tmp/duplicate/000: renamed to /tmp/duplicate/00
shred: /tmp/duplicate/00: renamed to /tmp/duplicate/0
shred: /tmp/duplicate/file.mkv: removed
[100%] Done!
$ ls duplicate/

```

**Comment by sahib** [comment]
> I think I gave up here. It's obvious that I don't understand the usercmd handler :/

I'll try to add an example. Maybe even this one.

> rmlint -c 'sh:cmd=shred -vun 10 $1' works if the files don't contain special characters, just like spaces:

My bad, I forget the quotes around ``"$1"``.

```bash
rmlint -c 'sh:cmd=shred -un 10 "$1" && echo "Shredding $1"' /tmp/xxx 
```

> I also noticed if the file is completely regular (no special accented letters or spaces), the dry run ./rmlint -n is rather wet:

Hmm, you're kinda right. In dry mode it should not execute the command, need to fix that.
We currently do not have a separate check for user specific commands for a dry run.

The additional output is there because you put the ``-v`` flag there. With that ``shred`` will start to spam your output like crazy. The example above replaces that with a simple ``echo`` output.

**Comment by p1839251** [comment]
Thank you for your help!
Add that example to the manual please, it's very helpful.
Thanks

**Comment by sahib** [comment]
I've fixed as of ``836ed50c``. Should be all done now.
